### DISCUSSION TITLE
How to correctly use valhalla_assign_speeds?

### DISCUSSION THREAD

**Discussion by nvcnvn** [valhalla/valhalla#4307] [Q&A] [answered]
I have this default speed config file https://gist.github.com/nvcnvn/1fb4d43feabfbd674a52b1b80b515963
The file has 3 objects which I hope will override the data for my country (VN) and my city (SG) and the rest.
```
[
    {...},
    {
        "iso3166-1": "VN",
        ....
    },
    {
        "iso3166-1": "VN",
        "iso3166-2": "SG",
        ....
    }
]
```
Then I run "valhalla_assign_speeds --config valhalla.json" then run the web service.
From the logs it seem work:
```
2023/09/25 11:17:57.357373 [INFO] Running valhalla_assign_speeds with 4 thread(s).
2023/09/25 11:17:57.360882 [INFO] Enabled default speeds assignment from config: /custom_configs/default_speed.json
2023/09/25 11:17:57.360945 [INFO] Enabled default speeds assignment from config: /custom_configs/default_speed.json
2023/09/25 11:17:57.361011 [INFO] Enabled default speeds assignment from config: /custom_configs/default_speed.json
2023/09/25 11:17:57.365518 [INFO] Enabled default speeds assignment from config: /custom_configs/default_speed.json
2023/09/25 11:18:00.419430 [INFO] Assigned speeds to 11548382 edges in total out of 12052186
INFO: Found config file. Starting valhalla service!
2023/09/25 11:18:01.048785 [INFO] Tile extract successfully loaded with tile count: 737
2023/09/25 11:18:01.048851 [INFO] Tile extract successfully loaded with tile count: 737
2023/09/25 11:18:01.049949 [INFO] Tile extract successfully loaded with tile count: 737
2023/09/25 11:18:01.050018 [INFO] Tile extract successfully loaded with tile count: 737
2023/09/25 11:18:01.050090 [INFO] Tile extract successfully loaded with tile count: 737
2023/09/25 11:18:01.050285 [INFO] Tile extract successfully loaded with tile count: 737
2023/09/25 11:18:01.050752 [INFO] Tile extract successfully loaded with tile count: 737
2023/09/25 11:18:01.050840 [INFO] Tile extract successfully loaded with tile count: 737
```
But when I test, the speed for residential roads still around 30 kmh (like the hardcoded value).
https://www.openstreetmap.org/#map=18/10.77412/106.60446 
<img width="961" alt="Screenshot 2023-09-25 at 18 33 29" src="https://github.com/valhalla/valhalla/assets/1708320/34d47c40-4a3d-40f9-ac2e-b5deb95d492d">

Anything I can do for debug more about this issue?

**Comment by nilsnolde** [comment, marked as answer]
Well, that'd not update the tar file, you're not running `valhalla_build_extract` again. Once you do you should see the expected behavior.

**Reply by nilsnolde** [reply]
Is `SG` surely a "state"? You say city, but it needs to be a `iso3166-2` admin in OSM.

**Reply by nvcnvn** [reply]
It a city-state :D thing :D 
But to clarify, if we have an empty "iso3166-1" and "iso3166-2" key mean I already have a global speed config right?
I download the map data only for a single country and I only need to set the speed the same for everywhere.

**Reply by nvcnvn** [reply]
BTW, its seem my config file having some issue:
```
2023/09/26 03:17:18.246572 [WARN] Exceeding maximum.  Average speed: 141
2023/09/26 03:17:18.246833 [WARN] Exceeding maximum.  Average speed: 141
2023/09/26 03:17:18.247504 [WARN] Exceeding maximum.  Average speed: 141
```

**Reply by nvcnvn** [reply]
after running the same command again and removing the tiles folder, restart the service and it seem working
